spec/acceptance/fail2ban_spec.rb

```diff
@@ -0,0 +1,237 @@
+    it_behaves_like 'an idempotent resource' do
+      let(:manifest) do
+        <<-EOS
+      class { 'fail2ban': }
+        EOS
+      end
+    end
+
+    describe package(package_name) do
+      it { is_expected.to be_installed }
+    end
+
+    describe file(config_file_path) do
+      it { is_expected.to be_file }
+    end
+
+    describe service(service_name) do
+      it { is_expected.to be_running }
+      it { is_expected.to be_enabled }
+    end
+
+    describe command('fail2ban-client status') do
+      its(:stdout) { is_expected.to match %r{Jail list:\s+#{ssh_jail}} }
+    end
+  end
+
+  context 'when package absent' do
+    it_behaves_like 'an idempotent resource' do
+      let(:manifest) do
+        <<-EOS
+      class { 'fail2ban':
+        package_ensure => 'absent',
+        service_ensure => 'stopped',
+        service_enable => false,
+      }
+        EOS
+      end
+    end
+
+    describe package(package_name) do
+      it { is_expected.not_to be_installed }
+    end
+
+    describe file(config_file_path) do
+      it { is_expected.to be_file }
+    end
+
+    describe service(service_name) do
+      it { is_expected.not_to be_running }
+      it { is_expected.not_to be_enabled }
+    end
+  end
+
+  context 'when content template' do
+    it_behaves_like 'an idempotent resource' do
+      let(:manifest) do
+        <<-EOS
+      $_config_file_template = $facts['os']['family'] ? {
+        'RedHat' => "fail2ban/RedHat/#{fact('os.release.major')}/#{config_file_path}.epp",
+        default  => "fail2ban/#{fact('os.name')}/#{fact('os.release.major')}/#{config_file_path}.epp",
+      }
+      class { 'fail2ban':
+        config_file_template => $_config_file_template,
+      }
+        EOS
+      end
+    end
+
+    describe file(config_file_path) do
+      it { is_expected.to be_file }
+      it { is_expected.to contain 'THIS FILE IS MANAGED BY PUPPET' }
+      it { is_expected.to contain %r{^chain = INPUT$} }
+    end
+  end
+
+  context 'when content template and custom chain' do
+    it_behaves_like 'an idempotent resource' do
+      let(:manifest) do
+        <<-EOS
+      $_config_file_template = $facts['os']['family'] ? {
+        'RedHat' => "fail2ban/RedHat/#{fact('os.release.major')}/#{config_file_path}.epp",
+        default  => "fail2ban/#{fact('os.name')}/#{fact('os.release.major')}/#{config_file_path}.epp",
+      }
+      class { 'fail2ban':
+        config_file_template => $_config_file_template,
+        iptables_chain => 'TEST',
+      }
+        EOS
+      end
+    end
+
+    describe file(config_file_path) do
+      it { is_expected.to be_file }
+      it { is_expected.to contain 'THIS FILE IS MANAGED BY PUPPET' }
+      it { is_expected.to contain %r{^chain = TEST$} }
+    end
+  end
+
+  context 'when content template and custom banaction' do
+    it_behaves_like 'an idempotent resource' do
+      let(:manifest) do
+        <<-EOS
+      $_config_file_template = $facts['os']['family'] ? {
```

Comment:
Same issue with outdenting.  This would also be better as an `example/*.pp` so that it is linted.